hibernate-orm-compatibility-5.6 tests are failing on CompatibilityTest.offsetTime

### Describe the bug

hibernate-orm-compatibility-5.6 tests are failing on CompatibilityTest.offsetTime
Applies to both mariadb and postgresql module.

```
[ERROR] Failures:
[ERROR]   CompatibilityTest.offsetTime:122
expected: 11:58:30+01:00
 but was: 12:58:30+01:00
```

As it has issues with time shift from summer time to winter time.

Noticed on 3.2 branch too when checking https://github.com/quarkusio/quarkus/pull/36697

### Expected behavior

hibernate-orm-compatibility-5.6 tests pass on CompatibilityTest.offsetTime

### Actual behavior

hibernate-orm-compatibility-5.6 tests are failing on CompatibilityTest.offsetTime

### How to Reproduce?

 - Build Quarkus main
 - run `mvn clean verify -Dtest-containers -Dstart-containers -f integration-tests/hibernate-orm-compatibility-5.6/postgresql`
 - run `mvn clean verify -Dtest-containers -Dstart-containers -f integration-tests/hibernate-orm-compatibility-5.6/mariadb`

### Output of `uname -a` or `ver`

macOS

### Output of `java -version`

_No response_

### Quarkus version or git rev

Quarkus main 3149eced5c1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/integration-tests/hibernate-orm-compatibility-5.6/database-generator/src/main/java/io/quarkus/it/hibernate/compatibility/Main.java
+++ b/integration-tests/hibernate-orm-compatibility-5.6/database-generator/src/main/java/io/quarkus/it/hibernate/compatibility/Main.java
@@ -72,7 +72,7 @@ public class Main {
                 checkEqual(createdEntity.duration, loadedEntity.duration);
                 checkEqual(createdEntity.uuid, loadedEntity.uuid);
                 checkEqual(createdEntity.instant, loadedEntity.instant);
-                checkEqual(createdEntity.offsetTime.withOffsetSameInstant(ZoneId.systemDefault().getRules().getOffset(Instant.now())),
+                checkEqual(createdEntity.offsetTime.toLocalTime().atOffset(ZoneId.systemDefault().getRules().getOffset(Instant.now())),
                         loadedEntity.offsetTime);
                 checkEqual(createdEntity.offsetDateTime.atZoneSameInstant(ZoneId.systemDefault()).toOffsetDateTime(),
                         loadedEntity.offsetDateTime);

--- a/integration-tests/hibernate-orm-compatibility-5.6/mariadb/src/test/java/io/quarkus/it/hibernate/compatibility/CompatibilityTest.java
+++ b/integration-tests/hibernate-orm-compatibility-5.6/mariadb/src/test/java/io/quarkus/it/hibernate/compatibility/CompatibilityTest.java
@@ -120,9 +120,8 @@ public class CompatibilityTest {
     public void offsetTime() {
         assertThat(findOld().offsetTime)
                 .isEqualTo(LocalTime.of(12, 58, 30, 0)
-                        .atOffset(ZoneOffset.ofHours(2))
                         // Hibernate ORM 5 used to normalize these values to the JVM TZ
-                        .withOffsetSameInstant(ZoneId.systemDefault().getRules().getOffset(Instant.now())));
+                        .atOffset(ZoneId.systemDefault().getRules().getOffset(Instant.now())));
     }
 
     // https://github.com/hibernate/hibernate-orm/blob/6.2/migration-guide.adoc#timezone-and-offset-storage

--- a/integration-tests/hibernate-orm-compatibility-5.6/postgresql/src/test/java/io/quarkus/it/hibernate/compatibility/CompatibilityTest.java
+++ b/integration-tests/hibernate-orm-compatibility-5.6/postgresql/src/test/java/io/quarkus/it/hibernate/compatibility/CompatibilityTest.java
@@ -120,9 +120,8 @@ public class CompatibilityTest {
     public void offsetTime() {
         assertThat(findOld().offsetTime)
                 .isEqualTo(LocalTime.of(12, 58, 30, 0)
-                        .atOffset(ZoneOffset.ofHours(2))
                         // Hibernate ORM 5 used to normalize these values to the JVM TZ
-                        .withOffsetSameInstant(ZoneId.systemDefault().getRules().getOffset(Instant.now())));
+                        .atOffset(ZoneId.systemDefault().getRules().getOffset(Instant.now())));
     }
 
     // https://github.com/hibernate/hibernate-orm/blob/6.2/migration-guide.adoc#timezone-and-offset-storage

